### ISSUE TITLE
merging to upstream

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/crunch#2]
I move [the issue](https://github.com/Unvanquished/crunch/issues/1) there so we can remove the old unvanquished fork.

There was some PR made on BinomialLLC's side but they were not merged:

- [ ] BinomialLLC/crunch#12
- [x] BinomialLLC/crunch#13
  * https://github.com/BinomialLLC/crunch/pull/25 by someone else was merged instead
- [ ] BinomialLLC/crunch#14

Since they [officially stated](https://github.com/BinomialLLC/crunch/commit/235946f7a1cf8b9c97e8bf0e8062d5439a51dec7) they don't support crunch any more, it's likely to never happens.

Then Unity people took the code and improved it a lot, so I replayed the commits on their tree with hope they will upstream them, I also added another commit to fix an issue they introduced:

- [ ] Unity-Technologies/crunch#2
- [ ] Unity-Technologies/crunch#3
- [ ] Unity-Technologies/crunch#5
- [ ] Unity-Technologies/crunch#7
- [ ] Unity-Technologies/crunch#8
- [ ] Unity-Technologies/crunch#9
- [ ] Unity-Technologies/crunch#10
- [ ] Unity-Technologies/crunch#13

Basically the commits fix compilation on 64 bit, fix compilation on Linux, fix build outside of Visual Studio, and add an improvement: to renormalize on the top of mip-level too using the `-rtopmip` option with the `--renormalize` one, allow user-defined `CRND_ASSERT`, fix malloc on macOS and FreeBSD.

**Comment by illwieckz** [comment]
We stopped creating PR to upstream long time ago. None of ours were merged, and PR from others didn't get merged either.

Let's accept our fate, a merge to upsteam will never happen.
